app/src/main/java/app/revanced/manager/ui/viewmodel/InstallerViewModel.kt

```diff
@@ -215,6 +241,19 @@ class InstallerViewModel(
         }
     }
 
+    fun reinstall() = viewModelScope.launch {
+        try {
+            pm.getPackageInfo(outputFile)?.packageName?.let { pm.uninstallPackage(it) }
+                ?: throw Exception("Failed to load application info")
+
+            pm.installApp(listOf(outputFile))
+        } catch (e: Exception) {
+            Log.e(tag, "Failed to reinstall", e)
+            app.toast(app.getString(R.string.reinstall_app_fail, e.simpleMessage()))
+        }
+
```

Comment:
We have a function in `util/Util.kt`. I think it can be used here to deal with exceptions